keras_nlp/models/llama3/llama3_tokenizer.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.tokenizers.byte_pair_tokenizer import BytePairTokenizer
+
+
+@keras_nlp_export("keras_nlp.models.Llama3Tokenizer")
+class Llama3Tokenizer(BytePairTokenizer):
+    def __init__(
+        self,
+        vocabulary=None,
+        merges=None,
+        **kwargs,
+    ):
+        self.start_token = "<|begin_of_text|>"
+        self.end_token = "<|end_of_text|>"
+
+        super().__init__(
+            vocabulary=vocabulary,
+            merges=merges,
+            unsplittable_tokens=[self.start_token, self.end_token],
```

Comment:
Note: I noticed a bunch of reserved tokens that LLaMA 3 uses that should also be unsplittable. When I add them here though, I get a maximum recursion depth reached error. Not a huge blocker, we can document that reserved tokens aren't supported yet.

cc @mattdangerw